### ISSUE TITLE
Front Matter doesn't render in preview but is generated when exported to HTML

### ISSUE THREAD

**Issue by alaut** [yzhang-gh/vscode-markdown#937]
I appreciate the consistency between the VS Code markdown preview from this extension and the generated HTML. Ideally I'd like it if Front matter fields like author, date, title were both parsed and show in the VS Code preview & the HTML export.

Otherwise it seems the HTML output just exports the frontmatter as a one-liner which is ugly.

**Comment by yzhang-gh** [comment]
You can give [Markdown yaml Preamble](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-yaml-preamble) a try, which is from a VS Code team member.

**Comment by alaut** [comment]
this seams like a reasonable extension that will at least make my html renderings more reasonable. It still would be nice if one could include standard tags like author, date, title, etc. and the converter could parse this and generate a nice header as done in pandoc for example.

Maybe this is better suited for a feature request.

**Comment by yzhang-gh** [comment]
I'm not sure whether the above extension still works as expected. Does it show a table-like header as in its screenshot?

And can you give me an example/picture of the pandoc-style header?

**Comment by Lemmingh** [comment]
> standard tags

There is no standard, even no widely accepted convention.

> parse this and generate a nice header as done in pandoc for example

How such things are processed is tied to **specific build system**.

For example, pandoc has its own syntax and conversion template.

---

#238
#437

**Comment by alaut** [comment]
fair enough, but i think the rendering in the VS Code preview should match that of the exported html. It seems by default the YAML style front matter is ignored by the VS Code preview so I would think the corresponding html should do the same rather than rendering a table. 

If the table is necessary, maybe transposing it would make more sense?

**Comment by alaut** [comment]
> I'm not sure whether the above extension still works as expected. Does it show a table-like header as in its screenshot?

It does show a table like header. Personally I'm not a fan but maybe would find the table more useful if it was transposed, making it more readable.

> And can you give me an example/picture of the pandoc-style header?

As used in [Jekyl](https://jekyllrb.com/docs/front-matter/) and [Hugo](https://gohugo.io/content-management/front-matter/), and also in [Pandoc](https://pandoc.org/MANUAL.html#extension-yaml_metadata_block) using triple dashes.

```
---
author: Author
title: Title
subtitle: Subtitle
date: Today
---
````

**Comment by yzhang-gh** [comment]
> It seems by default the YAML style front matter is ignored by the VS Code preview so I would think the corresponding html should do the same rather than rendering a table.

Alright, I thought you would like to have a table-like header.

Then probably we need to replicate what VS Code does
https://github.com/microsoft/vscode/blob/94c9ea46838a9a619aeafb7e8afd1170c967bb55/extensions/markdown-language-features/src/markdownEngine.ts#L97-L110
